app/scripts/migrations/059.test.js

```diff
@@ -0,0 +1,384 @@
+import { strict as assert } from 'assert';
+import { cloneDeep } from 'lodash';
+import {
+  KOVAN_CHAIN_ID,
+  MAINNET_CHAIN_ID,
+  RINKEBY_CHAIN_ID,
+  GOERLI_CHAIN_ID,
+} from '../../../shared/constants/network';
+import { TRANSACTION_TYPES } from '../../../shared/constants/transaction';
+import migration59 from './059';
+
+const ERRONEOUS_TRANSACTION_STATE = {
```

Comment:
Can we add a case to these tests that where there is a transaction with a 'cancel' type, but it shares a nonce with a non-cancel transaction, to verify that it won't be removed?

Reply:
there is one. line 192

```js
 it('should not drop cancel transactions with matching non cancel or retry in same network and nonce', async function () {
    const oldStorage = {
      meta: {},
      data: {
        TransactionController: {
          transactions: {
            ...ERRONEOUS_TRANSACTION_STATE,
            11: {
              ...ERRONEOUS_TRANSACTION_STATE['0'],
              id: 11,
              type: TRANSACTION_TYPES.SENT_ETHER,
            },
          },
        },
        foo: 'bar',
      },
    };

    const newStorage = await migration59.migrate(oldStorage);
    assert.deepEqual(newStorage.data, {
      TransactionController: {
        transactions: oldStorage.data.TransactionController.transactions,
      },
      foo: 'bar',
    });
  });
```